File: app_pasos/lib/models/exercise.dart
class Exercise {
  final String id;
  final String name;
  final String nameSpanish;
  final String category;
  final String imageUrl;
  final int defaultSets;
  final String defaultReps;
  final int restTime;
  final String description;
  final String descriptionSpanish;
  final String videoUrl;
  final String muscle;
  final String equipment;
  final String difficulty;

  String get displayName => nameSpanish.isNotEmpty ? nameSpanish : name;
  String get displayDescription => descriptionSpanish.isNotEmpty ? descriptionSpanish : description;

  Exercise({
    required this.id,
    required this.name,
    required this.category,
    this.nameSpanish = '',
    this.imageUrl = '',
    this.defaultSets = 3,
    this.defaultReps = '10',
    this.restTime = 60,
    this.description = '',
    this.descriptionSpanish = '',
    this.videoUrl = '',
    this.muscle = '',
    this.equipment = '',
    this.difficulty = '',
  });

  factory Exercise.fromJson(Map<String, dynamic> json) {
    return Exercise(
      id: json['id']?.toString() ?? json['_id']?.toString() ?? '',
      name: json['name'] ?? '',
      nameSpanish: json['nameSpanish'] ?? '',
      category: json['category'] ?? 'strength',
      imageUrl: json['imageUrl'] ?? '',
      defaultSets: json['defaultSets'] ?? 3,
      defaultReps: json['defaultReps']?.toString() ?? '10',
      restTime: json['restTime'] ?? 60,
      description: json['description'] ?? '',
      descriptionSpanish: json['descriptionSpanish'] ?? '',
      videoUrl: json['videoUrl'] ?? '',
      muscle: json['muscle'] ?? '',
      equipment: json['equipment'] ?? '',
      difficulty: json['difficulty'] ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'nameSpanish': nameSpanish,
    'category': category,
    'imageUrl': imageUrl,
    'defaultSets': defaultSets,
    'defaultReps': defaultReps,
    'restTime': restTime,
    'description': description,
    'descriptionSpanish': descriptionSpanish,
    'videoUrl': videoUrl,
    'muscle': muscle,
    'equipment': equipment,
    'difficulty': difficulty,
  };
}

File: backend/models/Exercise.js
const mongoose = require('mongoose');

const exerciseSchema = new mongoose.Schema({
  externalId: {
    type: String,
    required: true,
    unique: true,
    index: true
  },
  source: {
    type: String,
    enum: ['exercisedb', 'wger'],
    required: true
  },
  name: {
    type: String,
    required: true
  },
  nameSpanish: {
    type: String,
    default: ''
  },
  category: {
    type: String,
    enum: ['strength', 'cardio', 'warmup', 'flexibility'],
    default: 'strength'
  },
  bodyPart: {
    type: String,
    default: ''
  },
  target: {
    type: String,
    default: ''
  },
  equipment: {
    type: String,
    default: ''
  },
  gifUrl: {
    type: String,
    default: ''
  },
  imageUrl: {
    type: String,
    default: ''
  },
  localImage: {
    type: Buffer,
    default: null,
  },
  localImageMime: {
    type: String,
    default: '',
  },
  instructions: [{
    type: String
  }],
  instructionsSpanish: [{
    type: String
  }],
  description: {
    type: String,
    default: ''
  },
  descriptionSpanish: {
    type: String,
    default: ''
  },
  defaultSets: {
    type: Number,
    default: 3
  },
  defaultReps: {
    type: String,
    default: '10'
  },
  restTime: {
    type: Number,
    default: 60
  }
}, {
  timestamps: true
});

exerciseSchema.index({ name: 'text', nameSpanish: 'text' });
exerciseSchema.index({ category: 1, name: 1 });
exerciseSchema.index({ source: 1, externalId: 1 });

module.exports = mongoose.model('Exercise', exerciseSchema);
